### PR TITLE
DDT-752: RequiredIfTrue validator

### PR DESCRIPTION
**Pull request by mp-assco**
Usage:

```
@Data
@RequiredIfTrue.List({
  @RequiredIfTrue(isFieldReferenceRequired = "isImportLicenseRequired", fieldReference = "importLicenseReference"),
  @RequiredIfTrue(isFieldReferenceRequired = "isExportDeclarationRequired", fieldReference = "exportDeclarationReference")
})
public abstract class AbstractBookingTO {
  ...
  protected Boolean isExportDeclarationRequired;

  protected String exportDeclarationReference;

  protected Boolean isImportLicenseRequired;

  protected String importLicenseReference;
  ...
}

```

### CODE REVIEW

File: src/main/java/org/dcsa/core/validator/RequiredIfTrue.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+@Constraint(validatedBy = RequiredIfTrueValidator.class)
+public @interface RequiredIfTrue {
+
+    String isFieldReferenceRequired();
+
+    String fieldReference();
+
+    boolean allowNull() default false;
+
+    String message() default "Ding dong dillermand!";
```
Comment (nt-as) [inline src/main/java/org/dcsa/core/validator/RequiredIfTrue.java:24]: Better default message please.

**Review by nt-as — CHANGES_REQUESTED** [review]

File: src/main/java/org/dcsa/core/validator/RequiredIfTrue.java (hunk 1 above)
Comment (mp-assco) [reply src/main/java/org/dcsa/core/validator/RequiredIfTrue.java]: Fair enough

File: src/main/java/org/dcsa/core/validator/RequiredIfTrue.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.*;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = RequiredIfTrueValidator.class)
+public @interface RequiredIfTrue {
```
Comment (nt-as) [inline src/main/java/org/dcsa/core/validator/RequiredIfTrue.java:16]: Adding `@Repeatable(RequiredIfTrue.List.class)` here might make it possible to make the list annotation implicit (will require that `@Target`, `@Retention` and `@Documented` annotation is copied on to the `RequiredIfTrue.List` class.

File: src/main/java/org/dcsa/core/validator/RequiredIfTrue.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    String isFieldReferenceRequired();
+
+    String fieldReference();
+
+    boolean allowNull() default false;
```
Comment (nt-as) [inline src/main/java/org/dcsa/core/validator/RequiredIfTrue.java:22]: I would remove this.  Where relevant, please can use `@NotNull` instead, which gives a better error message.  Replace it with `true` in the rest of the code will suffice.

**Pull request closed without merging**